router.js

```diff
@@ -0,0 +1,104 @@
+const express = require("express");
+const router = express.Router();
+const dbHelper = require("./dbHelper");
+const encrypt = require("bcryptjs");
+const auth = require("./authMiddleware");
+
+router.get("/", (req, res) => {
+  res.status(200).json({ message: "the router is working" });
+});
+
+router.post("/register", (req, res) => {
+  const hash = encrypt.hashSync(req.headers.password, 10);
+  req.session.username = req.headers.username;
+  req.session.password = hash;
```

Comment:
i would not add the password to a session or a jwt, there is no reason to every return the hash to the frontend